python_modules/dagster/dagster/_config/pythonic_config/conversion_utils.py

```diff
@@ -251,6 +175,9 @@ def _config_type_for_type_on_pydantic_field(
     Args:
         potential_dagster_type (Any): The Python type of the Pydantic field.
     """
+    while get_origin(potential_dagster_type) == Annotated:
+        potential_dagster_type = get_args(potential_dagster_type)[0]
+
```

Comment:
nit: extract this into its own function could make it clearer what is going on.

`strip_all_wrapping_annotated_types` or something